backend/src/api/model/series.rs

```diff
@@ -99,6 +103,156 @@ impl Series {
+        } else if old_realm.name_from_block.is_some()
+            && old_realm.name_from_block == blocks[0].id().key_for(Id::BLOCK_KIND)
+        {
+            // The realm has its name derived from the series block that is being removed - so the name
+            // shouldn't be used anymore. Ideally this would restore the previous title,
+            // but that isn't stored anywhere. Instead the realm is given the name of its path segment.
+            Realm::rename(
+                old_realm.id(),
+                UpdatedRealmName::plain(old_realm.path_segment),
+                context,
+            ).await?;
+        }
+
+        let removed_block = BlockValue::remove(blocks[0].id(), context).await?;
+
+        Ok(RemoveMountedSeriesOutcome::RemovedBlock(removed_block))
+    }
+
+    pub(crate) async fn mount(
+        series: NewSeries,
+        parent_realm_path: String,
+        new_realms: Vec<RealmSpecifier>,
+        context: &Context,
+    ) -> ApiResult<Realm> {
+        context.auth.required_trusted_external()?;
+
+        // Store series information in DB.
+        let series = Self::announce(series, context).await?;
+
+        // Create realms if necessary.
+        let mut realm_components = Vec::new();
+        let parts = parent_realm_path
+            .split('/')
+            .filter(|s| !s.is_empty())
+            .collect::<Vec<&str>>();
+        
+        let mut current_path = String::from("/");
+        let mut target_path = String::from("/");
+        for part in parts.iter() {
+            current_path.push_str(part);
+            current_path.push('/');
+
+            target_path.push_str(&current_path);
+
+            let realm = Realm::load_by_path(current_path.clone(), context)
+                .await?
+                .ok_or_else(|| invalid_input!("`path` {} does not refer to a valid realm", current_path))?;
+
+            realm_components.push(RealmLineageComponent {
+                // The name field in a `RealmSpecifier` is `Option<String>`,
+                // but `RealmLineageComponent` needs this to be a `String`.
+                // This can only be `None` for the actual target realm, in which case the
+                // name will be overwritten anyway with the series name, so no realm
+                // will actually end up being named "dummy".
+                name: realm.resolved_name.unwrap_or("dummy".to_string()),
+                path_segment: realm.path_segment,
+            });
+        }
+
+        for RealmSpecifier { name, path_segment } in new_realms {
+            target_path.push_str(&path_segment.clone());
+            realm_components.push(RealmLineageComponent {
+                name: name.unwrap_or("dummy".to_string()),
+                path_segment,
+            });
+        }
+
+        Realm::create_lineage(realm_components, context).await?;
+
+        // Add series block in target realm.
+        Self::add_mount_point(series.opencast_id, target_path, context).await
```

Comment:
That's a lot of code converting one parameter to another. So I have to retract my previous suggestion and agree with you: this part does not need to use `create_lineage`, we are better off just doing it like before. We can use the other functions tho. I think this should work (untested tho):

```rust
context.auth.required_trusted_external()?;

// Check parameters
if new_realms.iter().rev().skip(1).any(|r| r.name.is_none()) {
    return Err(invalid_input!("all new realms except the last need to have a name"));
}

let parent_realm = Realm::load_by_path(parent_realm_path, context)
    .await?
    .ok_or_else(|| invalid_input!("`parentRealmPath` does not refer to a valid realm"))?;

if new_realms.is_empty() {
    let blocks = BlockValue::load_for_realm(parent_realm.key, context).await?;
    if !blocks.is_empty() {
        return Err(invalid_input!("series can only be mounted in empty realms"));
    }
}

// Create series
let series = Series::create(series, context).await?;

// Create realms
let target_realm = {
    let mut target_realm = parent_realm;
    for RealmSpecifier { name, path_segment } in new_realms {
        target_realm = Realm::add(NewRealm {
            // The `unwrap_or` case is only potentially used for the
            // last realm, which is renamed below anyway. See the check
            // above.
            name: name.unwrap_or_else(|| "temporary-dummy-name".into()),
            path_segment,
            parent: Id::realm(target_realm.key),
        }, context).await?
    }
    target_realm
};

// Create mount point
Series::add_mount_point(series.opencast_id, target_realm.path, context).await
```